rs/ic_os/partition_tools/src/ext.rs

```diff
@@ -230,6 +206,48 @@ impl ExtPartition {
 
         Ok(())
     }
+
+    fn check_debugfs_result(out: &Output) -> Result<()> {
+        let errors = std::str::from_utf8(&out.stderr)?
+            .lines()
+            .filter(|error| !error.starts_with("debugfs")) // version number
+            // Some errors we ignore.
+            .filter(|error| !error.contains("Ext2 directory already exists"))
+            .filter(|error| !error.contains("rm: File not found"))
+            .join("\n");
+        if !out.status.success() || !errors.is_empty() {
+            bail!("debugfs failed:\n{errors}");
+        }
+
+        Ok(())
+    }
+
+    fn debugfs_input(input: &Path, output: &Path) -> String {
+        // Commands that generate all parent dirs of input (debugfs doesn't
+        // support mkdir -p)
+        // Eg. if `output` is `/opt/ic/bin/something.txt`, `mkdir_all` will be:
+        // mkdir /opt
+        // mkdir /opt/ic
+        // mkdir /opt/ic/bin
+        let mkdir_all = std::iter::successors(output.parent(), |child| child.parent())
+            .filter(|path| *path != Path::new("/"))
+            .collect::<Vec<_>>()
+            .into_iter()
+            .rev()
+            .map(|path| format!("mkdir {}", path.display()))
+            .join("\n");
```

Comment:
Optional nit, just a bit more readable IMO:
```
let mkdir_all = output
        .ancestors() 
        .take_while(|path| path != &Path::new("/")) 
        .collect::<Vec<_>>()
        .iter()
        .rev()
        .map(|path| format!("mkdir {}", path.display()))
        .collect::<Vec<_>>()
        .join("\n");
```

Reply:
I didn't know of `ancestors`, this is great!